src/lib/grad.jl

```diff
@@ -14,3 +14,27 @@ macro which(ex)
   @capture(ex, f_(args__)) || error("Zygote.@which f(args...)")
   :(InteractiveUtils.@which adjoint(Context(), $(esc(f)), $(esc.(args)...)))
 end
+
+"""
+
+    checkpointed(f, xs...)
+
+Use gradient checkpointing on the call `f(xs...)`. This means that
+`checkpointed(f, xs...) === f(xs...)`, but when computing the derivative
+intermediate results from the forward pass of `f` will not be stored. Instead the forward
+pass will be repeated, when computing the derivative.
+This saves memory at the cost of increasing exectution time.
+
+!!! Warning:
+If `f` is not a pure function, `checkpointed` will likely give wrong results.
```

Comment:
```suggestion
!!! warning
    If `f` is not a pure function, `checkpointed` will likely give wrong results.
```
Right now this produces
```
help?> checkpointed
search: checkpointed

   checkpointed(f, xs...)

  Use gradient checkpointing on the call f(xs...). This means that checkpointed(f, xs...) ===
  f(xs...), but when computing the derivative intermediate results from the forward pass of f will
  not be stored. Instead the forward pass will be repeated, when computing the derivative. This
  saves memory at the cost of increasing exectution time.

  !!! warning: If f is not a pure function, checkpointed will likely give wrong results.
```
while with the suggestion it should produce (with the label in colour)
```
  not be stored. Instead the forward pass will be repeated, when computing the derivative. This
  saves memory at the cost of increasing exectution time.

  │ Warning
  │
  │  If f is not a pure function, checkpointed will likely give wrong results.

julia> 
```
which I think becomes a pretty box in HTML docs. 

Nice PR, btw!